frontend/public/components/namespace.jsx

```diff
@@ -530,25 +532,44 @@ const getProjectSelectedColumns = ({ showMetrics, showActions }) => {
 const ProjectLink = connect(null, {
   filterList: k8sActions.filterList,
 })(({ project, filterList }) => {
-  const [, setActiveNamespace] = useActiveNamespace();
+  const [, setLastNamespace] = useUserSettingsCompatibility(
+    LAST_NAMESPACE_NAME_USER_SETTINGS_KEY,
+    LAST_NAMESPACE_NAME_LOCAL_STORAGE_KEY,
+  );
+  const url = new URL(window.location.href);
+  const params = new URLSearchParams(url.search);
+  const basePath = url.pathname;
+  if (params.has('project-name')) {
+    // clear project-name query param from the url
+    params.delete('project-name');
+  }
+  const newUrl = {
+    search: `?${params.toString()}`,
+    hash: url.hash,
+  };
+  const namespacedPath = UIActions.formatNamespaceRoute(project.metadata.name, basePath, newUrl);
+
+  const handleClick = () => {
+    setLastNamespace(project.metadata.name);
+    // update last namespace in session storage (persisted only for current browser tab). Used to remember/restore if
+    // "All Projects" was selected when returning to the list view (typically from details view) via breadcrumb or
+    // sidebar navigation
+    sessionStorage.setItem(LAST_NAMESPACE_NAME_LOCAL_STORAGE_KEY, project.metadata.name);
+    // clear project-name filter when active namespace is changed
+    filterList(referenceForModel(ProjectModel), 'project-name', '');
+  };
+
   return (
     <span className="co-resource-item co-resource-item--truncate">
       <ResourceIcon kind="Project" />
-      <Button
-        isInline
+      <Link
+        to={namespacedPath}
         title={project.metadata.name}
```

Comment:
Can we remove the `title` prop? Don't see a reason for a tooltip with the exact same content. Also mentioned that in the ticket.

Reply:
Done.